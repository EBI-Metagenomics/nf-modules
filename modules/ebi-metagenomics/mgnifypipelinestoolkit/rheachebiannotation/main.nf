process MGNIFYPIPELINESTOOLKIT_RHEACHEBIANNOTATION {
    tag "${meta.id}"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:0.2.1--pyhdfd78af_0':
        'biocontainers/mgnify-pipelines-toolkit:0.2.1--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(fasta), path(diamond_tsv)
    path(rhea2chebi)

    output:
    tuple val(meta), path("*.tsv"), emit: proteins2rhea_tsv
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def is_fasta_compressed = fasta.name.endsWith(".gz")
    def is_diamond_compressed = diamond_tsv.name.endsWith(".gz")
    def fasta_file = fasta.name.replace(".gz", "")
    def diamond_file = diamond_tsv.name.replace(".gz", "")
    """
    if [ "$is_fasta_compressed" == "true" ]; then
        gzip -c -d $fasta > ${fasta_file}
    fi

    if [ "$is_diamond_compressed" == "true" ]; then
        gzip -c -d $diamond_tsv > ${diamond_file}
    fi

    add_rhea_chebi_annotation \\
        --diamond_hits ${diamond_file} \\
        --proteins ${fasta_file} \\
        --rhea2chebi ${rhea2chebi} ${args} \\
        --output ${prefix}.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """
}
