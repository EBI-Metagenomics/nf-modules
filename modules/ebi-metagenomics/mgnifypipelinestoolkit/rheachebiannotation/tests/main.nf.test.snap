{
    "add the rhea mapping based on the diamond uniref90 annotations - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,88927e8f6c9853475db255f3f08b6c30"
                ],
                "proteins2rhea_tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,88927e8f6c9853475db255f3f08b6c30"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-24T16:34:35.398436"
    },
    "add the rhea mapping based on the diamond uniref90 annotations": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,cbc725c9af469858b8ed48eb23ebbd40"
                    ]
                ],
                "1": [
                    "versions.yml:md5,88927e8f6c9853475db255f3f08b6c30"
                ],
                "proteins2rhea_tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,cbc725c9af469858b8ed48eb23ebbd40"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,88927e8f6c9853475db255f3f08b6c30"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-24T16:34:32.233299"
    }
}